Use correct width for codeblocks

Mister00X opened the pull request:
The previous settings for the `listings` package use `\textwidth` as line width
but `\textwidth` is "The full horizontal width of the entire page body."
which is too large in (nested) lists. Therefore use `\linewidth` which is
always the width of the current environment for the `listings`
environment.

Tested by generating the German manual.

Fixes siduction/sidu-manual#44

Mister00X committed:
Use correct width for codeblocks

The previous settings for the listings package use \textwidth as line width
but \textwidth is "The full horizontal width of the entire page body."
which is too large in (nested) lists. Therefore use \linewidth which is
always the width of the current environment for the listings
environment.

Tested by generating the German manual.

Fixes siduction/sidu-manual#44

hhl commented:
One question, should we shorten the code block a little,

`linewidth=.95\linewidth`,

 So, the code box is a bit smaller

Mister00X commented:
> One question, should we shorten the code block a little,
> 
> `linewidth=.95\linewidth`,
> 
> So, the code box is a bit smaller

Good idea :+1:  this may look visually more appealing.

I'm gonna change that.

EDIT: It looks indeed way better!

Mister00X committed:
Reduce line width of code blocks a bit

Reduce the line width of code blocks to 0.95\linewidth it looks more
visually appealing that way.

Thanks to @hhl for this idea!

Tested by generating the German manual.

hhl commented:
I have tested it too, works like charm.

Thank you for helping to bring the manual in form and fixing the LaTeX code.

Mister00X commented:
You're welcome. I am glad that I'm able to help siduction.
